boa/__init__.py

```diff
@@ -63,7 +63,15 @@ def __exit__(self, *args):
         set_env(self.old_env)
 
 
-def fork(url: str, block_identifier: int | str, **kwargs):
+def fork(
+    url: str, block_identifier: int | str = "safe", allow_dirty: bool = False, **kwargs
+):
+    global env
+    if env.evm.is_state_dirty and not allow_dirty:
```

Comment:
This env isn't used at all in this function. This code should be in `env.fork`

Reply:
It is, and no it should not

Reply:
I don't think this is the issue @AlbertoCentonze was trying to fix

Reply:
If it works on a new env why do we need the exception? I thought the issue was that users end up with an env with dirty pyevm and a half forked DB?

Reply:
No, boa.env.fork() always cleared the db. The issue is more that users might mix up order of operations and write to the state, then fork, e.g.
```
boa.loads(...)
boa.fork()
```
Then the state was written pre-fork() operation is zonked, which is confusing for the user.

Reply:
Alright then, thanks for the clarification